project-code/app/com/edulify/modules/geolocation/GeolocationFactory.java

```diff
@@ -0,0 +1,34 @@
+package com.edulify.modules.geolocation;
+
+/**
+ * Created by sowhat
+ */
+public class GeolocationFactory {
```

Comment:
As I commented before - it's to have an ablity to make isolated unit tests and have a control over instantiation of Geolocation object (for example in tests)
Also this decomposition is a way to follow the Single Responsibility principle